sdk/eventhub/Microsoft.Azure.EventHubs/tests/Client/ClientNegativeCases.cs

```diff
@@ -16,94 +16,118 @@ public class ClientNegativeCases : ClientTestBase
+                // Try sending.
+                PartitionSender sender = null;
+                await Assert.ThrowsAsync<MessagingEntityNotFoundException>(async () =>
+                {
+                    TestUtility.Log("Sending an event to nonexistent entity.");
+                    sender = ehClient.CreatePartitionSender("0");
+                    await sender.SendAsync(new EventData(Encoding.UTF8.GetBytes("this send should fail.")));
+                    throw new InvalidOperationException("Send call should have failed");
+                });
+                await sender.CloseAsync();
 
-            // Try receiving.
-            PartitionReceiver receiver = null;
-            await Assert.ThrowsAsync<MessagingEntityNotFoundException>(async () =>
+                // Try receiving.
+                await Assert.ThrowsAsync<MessagingEntityNotFoundException>(async () =>
+                {
+                    TestUtility.Log("Receiving from nonexistent entity.");
+                    receiver = ehClient.CreateReceiver(PartitionReceiver.DefaultConsumerGroupName, "0", EventPosition.FromStart());
+                    await receiver.ReceiveAsync(1);
+                    throw new InvalidOperationException("Receive call should have failed");
+                });
+                await receiver.CloseAsync();
+            }
+            finally
             {
-                TestUtility.Log("Receiving from nonexistent entity.");
-                receiver = ehClient.CreateReceiver(PartitionReceiver.DefaultConsumerGroupName, "0", EventPosition.FromStart());
-                await receiver.ReceiveAsync(1);
-                throw new InvalidOperationException("Receive call should have failed");
-            });
-            await receiver.CloseAsync();
+                await ehClient.CloseAsync();
+            }
 
             // Try receiving on an nonexistent consumer group.
             ehClient = EventHubClient.CreateFromConnectionString(TestUtility.EventHubsConnectionString);
-            await Assert.ThrowsAsync<MessagingEntityNotFoundException>(async () =>
+
+            try
             {
-                TestUtility.Log("Receiving from nonexistent consumer group.");
-                receiver = ehClient.CreateReceiver(Guid.NewGuid().ToString(), "0", EventPosition.FromStart());
-                await receiver.ReceiveAsync(1);
-                throw new InvalidOperationException("Receive call should have failed");
-            });
-            await receiver.CloseAsync();
+                await Assert.ThrowsAsync<MessagingEntityNotFoundException>(async () =>
+                {
+                    TestUtility.Log("Receiving from nonexistent consumer group.");
+                    receiver = ehClient.CreateReceiver(Guid.NewGuid().ToString(), "0", EventPosition.FromStart());
+                    await receiver.ReceiveAsync(1);
+                    throw new InvalidOperationException("Receive call should have failed");
```

Comment:
This is not required. Assert would fail on it's own,

Reply:
Nice catch.  I missed that when converting.